src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -2550,15 +2552,17 @@ private void checkDefendingPlanesCanLand() {
   }
 
   static CompositeChange clearTransportedByForAlliedAirOnCarrier(final Collection<Unit> attackingUnits,
-      final PlayerID attacker, final GameData data) {
+      final Territory battleSite, final PlayerID attacker, final GameData data) {
     final CompositeChange change = new CompositeChange();
     // Clear the transported_by for successfully won battles where there was an allied air unit held as cargo by an
     // carrier unit
     final Collection<Unit> carriers = Matches.getMatches(attackingUnits, Matches.unitIsCarrier());
     if (!carriers.isEmpty() && !Properties.getAlliedAirIndependent(data)) {
       final Predicate<Unit> alliedFighters = Match.allOf(Matches.isUnitAllied(attacker, data),
           Matches.unitIsOwnedBy(attacker).negate(), Matches.unitIsAir(), Matches.unitCanLandOnCarrier());
-      final Collection<Unit> alliedAirInTerr = Matches.getMatches(attackingUnits, alliedFighters);
+      final Collection<Unit> alliedAirInTerr = Matches.getMatches(
+          CollectionUtils.union(attackingUnits, battleSite.getUnits()),
```

Comment:
I could have done this with pure JDK code:

```java
Stream.concat(attackingUnits.stream(), battleSite.getUnits().stream).collect(Collectors.toSet())
```

But the requirement to use `toSet()` seemed buried in there.  So instead of adding a comment calling out the need to remove duplicate units, I just used Apache Collections' `union()`, which does this automatically (assuming the source collections themselves don't contain duplicates).